### DISCUSSION TITLE
Can I import *.json to *.mdx file in nextjs

### DISCUSSION THREAD

**Discussion by tienlx97** [mdx-js/mdx#2206] [Q&A] [answered]
I use nextjs + mdx to write blog. This is my folder contain all mdx file:
content
&nbsp;&nbsp;&nbsp;blog
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hello-mdx <-- folder
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;index.md
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;data.json

In **index.md** I have 2 components `<Comp1 />` and `<Comp2 />`: 

``` html
<Comp1>
   <Item key="pikachu" type="light" />
   <Item key="raichu" type="light" />
   <Item key="chikorita" type="leaf" />
   <Item key="mews" type="physic" />
</Comp1>

<Comp2>
   <Item2 append="1" key="pikachu" type="light" />
   <Item2 append="2" key="raichu" type="light" />
   <Item2 append="3" key="chikorita" type="leaf" />
   <Item2 append="4" key="mews" type="physic" />
</Comp2>
```
Because data in `<Comp1 />` and `<Comp2 />` are *static*. So I create a json file name **data.json**

``` json
{
 "pikachu": "light",
 "raichu": "light",
 "chikorita": "leaf"
}
```

Can I import **data.json** to **index.md** ? In **index.md** I should change 
``` html
<Comp1 jsonName="data" /> 
<Comp2 jsonName="data" />
```

**Comment by wooorm** [comment, marked as answer]
Needs more info: this is way too short.

Likely not though: you can't import json yet in ESM. In the future you could use import assertions, but they're not supported in most places yet.

**Reply by tienlx97** [reply]
sorry, I updated the information

**Reply by ChristianMurphy** [reply]
Adding on.
This is more a Next question, than an MDX one.
MDX will generate valid import syntax, it is up to the loader/runtime/platform to interpret it (in this case Next).

Next can load JSON files, though not through imports, see https://www.kindacode.com/article/next-js-reading-and-display-data-from-a-local-json-file/

And follow discussions of supporting ESM JSON in the next discussion forum.
https://github.com/vercel/next.js/discussions?discussions_q=esm+json+import
